### PR TITLE
Setting things up

### PR DESCRIPTION
**Pull request by jacobqvist**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,27 +1,93 @@
 {
-  "name": "web-template",
+  "name": "web",
```
Comment (jstroem) [inline package.json]: `@toitlang/wizard`

File: src/App.tsx [hunk 2]
```diff
@@ -1,20 +1,17 @@
+import { MuiThemeProvider } from "@material-ui/core";
 import React from "react";
-import logo from "./logo.svg";
 import "./App.css";
+import theme from "./assets/theme/theme";
+import MainView from "./components/MainView";
+
+// const serviceClient: ServiceClient = new ProtoServiceClient(API_URL);
+// export const context = new ServiceContext(serviceClient);
```
Comment (jstroem) [inline src/App.tsx]: remove

File: src/actions/wizard.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  SETUP = 2,
+  FLASH = 3,
+  DONE = 4,
+}
+
+export interface FlashingProperties {
+  ssid: string;
+  password: string;
+  firmware: string;
+  autoClaim: boolean;
```
Comment (jstroem) [inline src/actions/wizard.ts]: remove

File: src/actions/wizard.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  CONNECT = 1,
+  SETUP = 2,
+  FLASH = 3,
+  DONE = 4,
+}
+
+export interface FlashingProperties {
+  ssid: string;
+  password: string;
+  firmware: string;
```
Comment (jstroem) [inline src/actions/wizard.ts]: should we call this `firmware_version` ?

File: src/actions/wizard.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export enum WizardErrorType {
+  //CONNECT AND DETECT
+  UNKNOWN_ERR = "Contact support",
+  CREATE_PARTITION_ERR = "Create partition error",
+  //Flash and reinstall
+  FLASH_ERR = "Failed to flash",
+  //Claim
+  CLAIM_ERR = "Failed to claim",
```
Comment (jstroem) [inline src/actions/wizard.ts]: Remove

File: src/actions/wizard.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  type: UPDATE_FLASHING_PROPERTIES,
+  payload: { flashingProperties: flashingProperties },
+});
+
+export type UpdateShowQuestionnaire = {
+  type: typeof UPDATE_SHOW_QUESTIONNAIRE;
+  payload: { showQuestionnaire: boolean };
+};
+
+export const updateShowQuestionnaire = (showQuestionnaire: boolean): UpdateShowQuestionnaire => ({
```
Comment (jstroem) [inline src/actions/wizard.ts]: remove questionaire

File: src/actions/wizard.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  type: typeof UPDATE_FLASHING_PROPERTIES;
+  payload: { flashingProperties: FlashingProperties };
+};
+
+export const updateFlashingProperties = (flashingProperties: FlashingProperties): UpdateFlashingProperties => ({
+  type: UPDATE_FLASHING_PROPERTIES,
+  payload: { flashingProperties: flashingProperties },
+});
+
+export type UpdateShowQuestionnaire = {
```
Comment (jstroem) [inline src/actions/wizard.ts]: remove questionaire

File: src/components/MainView.tsx [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+const drawerWidthMobile = "100%";
+const gridSpacing = 8;
+const appBarHeight = 64;
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      display: "flex",
+    },
+    appBar: {
```
Comment (jstroem) [inline src/components/MainView.tsx]: do we have an appBar? if not remove.

File: src/components/MainView.tsx [hunk 9]
```diff
@@ -0,0 +1,283 @@
...
+      //   duration: theme.transitions.duration.enteringScreen,
+      // }),
+    },
+    drawerHeader: {
+      display: "flex",
+      alignItems: "center",
+      ...theme.mixins.toolbar,
+      height: appBarHeight,
+    },
+    drawer: {
```
Comment (jstroem) [inline src/components/MainView.tsx]: do we have a drawer? if not remove.

File: src/components/wizard/FlashView.tsx [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+    //     hardwareIdentity: {
+    //       hardwareID: identity.getHardwareId(),
+    //       model: properties.model,
+    //       name: "",
+    //     },
+    //     hardwareValid: true,
+    //   });
+    // } else {
+    //   this.props.updateDetectState(detectState);
+    // }
```
Comment (jstroem) [inline src/components/wizard/FlashView.tsx]: remove all of this add a todo:
```
// TODO: Fetch partitions.
```

File: src/components/wizard/images/claim_focus.svg [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<svg width="51" height="50" viewBox="0 0 51 50" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (jstroem) [inline src/components/wizard/images/claim_focus.svg]: remove claim icons

File: src/components/wizard/images/claim.svg [hunk 12]
```diff
@@ -0,0 +1,11 @@
+<svg width="51" height="50" viewBox="0 0 51 50" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (jstroem) [inline src/components/wizard/images/claim.svg]: remove claim icons

File: src/fonts.css [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+  @font-face {
+    font-family: 'ClashDisplay-Bold';
+    src: url('/fonts/ClashDisplay-Bold.eot');
+    src: url('/fonts/ClashDisplay-Bold.woff2') format('woff2'),
+         url('/fonts/ClashDisplay-Bold.woff') format('woff'),
+         url('/fonts/ClashDisplay-Bold.ttf') format('truetype');
+         font-weight: 700;
+         font-display: swap;
+         font-style: normal;
+  }
```
Comment (jstroem) [inline src/fonts.css]: could we ues https://www.cdnfonts.com/clash-display.font instead? if not we should have our licenses in place for the ClashDisplay font.

File: src/misc/serial/toitserial.ts [hunk 14]
```diff
@@ -0,0 +1,202 @@
+import "../../service/proto/uuid";
```
Comment (jstroem) [inline src/misc/serial/toitserial.ts]: this file only contains things that works in together with the console, remove the file

File: src/reducers/wizard.ts [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+      return {
+        ...state,
+        flashingProperties: action.payload.flashingProperties,
+      };
+    case UPDATE_CURRENT_ACTION:
+      return {
+        ...state,
+        currentAction: action.payload.currentAction,
+      };
+    case UPDATE_SHOW_QUESTIONNAIRE:
```
Comment (jstroem) [inline src/reducers/wizard.ts]: remove

File: src/reducers/wizard.ts [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+export type WizardState = {
+  readonly currentAction: WizardAction | WizardError;
+  readonly flashingProperties?: FlashingProperties;
+  readonly showQuestionnaire: boolean;
+};
+
+export const initialWizardState: WizardState = {
+  currentAction: WizardAction.CONNECT,
+  showQuestionnaire: false,
```
Comment (jstroem) [inline src/reducers/wizard.ts]: remove

File: src/reducers/wizard.ts [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+  UPDATE_SHOW_QUESTIONNAIRE,
+  WizardAction,
+  WizardError,
+  WizardReduxAction,
+} from "../actions/wizard";
+
+export type WizardState = {
+  readonly currentAction: WizardAction | WizardError;
+  readonly flashingProperties?: FlashingProperties;
+  readonly showQuestionnaire: boolean;
```
Comment (jstroem) [inline src/reducers/wizard.ts]: remove

**Review by jstroem — APPROVED** [review]

File: src/actions/wizard.ts (hunk 4 above)
Comment (jacobqvist) [reply src/actions/wizard.ts]: si

**Pull request merged**